src/svg/axis.js

```diff
@@ -0,0 +1,211 @@
+                var tickFormatter = tickFormat == null ? tryApply('tickFormat', fc.util.fn.identity) : tickFormat;
+
+                // orientation logic
+                var sign = orient === 'bottom' || orient === 'right' ? 1 : -1;
+                var pathTranspose = isVertical() ? transposeArray : fc.util.fn.identity;
+                var transform = isVertical() ? translateTransposed : translate;
+
+                var container = d3.select(this);
+
+                // add the domain line
+                var range = fc.util.scale.range(scale);
+                var domainPathData = pathTranspose([
+                      [range[0], sign * outerTickSize],
+                      [range[0], 0],
+                      [range[1], 0],
+                      [range[1], sign * outerTickSize]
+                    ]);
+
+                var domainLine = container.selectAll('path.domain')
+                    .data([data]);
+
+                domainLine.enter()
+                    .append('path')
+                    .classed('domain', true);
+
+                d3.transition(domainLine)
+                    .attr('d', svgDomainLine(domainPathData));
+
+                // datajoin and construct the ticks / label
+                dataJoin.attrs({
+                    'class': 'tick',
+                    // set the initial tick position based on the previous scale
+                    // in order to get the correct enter transition - however, for ordinal
+                    // scales the tick will not exist on the old scale, so use the current position
+                    'transform': containerTranslate(fc.util.scale.isOrdinal(scale) ? scale : scaleOld, transform)
+                });
+
+                var g = dataJoin(container, ticksArray);
+
+                // enter
+                var labelOffset = sign * (innerTickSize + tickPadding);
+                g.enter().classed('orient-' + orient, true);
+                g.enter().append('path');
+                g.enter()
+                    .append('text')
+                    .attr('transform', transform(0, labelOffset));
+
+                // update
+                g.attr('transform', containerTranslate(scale, transform));
+
+                g.selectAll('path')
+                    .attr('d', function(d) {
+                        return svgDomainLine(pathTranspose([
+                            [0, 0], [0, sign * innerTickSize]
+                        ]));
+                    });
+
+                g.selectAll('text')
+                   .attr('transform', transform(0, labelOffset))
+                   .text(tickFormatter);
+
+                // for non ordinal scales, exit by animating the tick to its new location
+                if (!fc.util.scale.isOrdinal(scale)) {
+                    g.exit()
+                        .attr('transform', containerTranslate(scale, transform));
+                }
+
+                decorate(g, data, index);
```

Comment:
I think you're missing the function to allow this to be configured? Does that mean there's no test of it?